### PR TITLE
feat/python: unlock slotindex

### PR DESCRIPTION
**Pull request by kwek20**
# Description of change

fixes [#963](https://github.com/iotaledger/iota-sdk/issues/963)

### CODE REVIEW

File: bindings/python/iota_sdk/types/unlock_condition.py [hunk 1]
```diff
@@ -79,9 +79,9 @@ class StorageDepositReturnUnlockCondition(UnlockCondition):
 class TimelockUnlockCondition(UnlockCondition):
     """A timelock unlock condition.
     Args:
-        unix_time: The Unix timestamp marking the end of the timelock.
+        slot_index: The slot index that determines when the associated output expires.
```
Comment (thibault-martinez) [inline bindings/python/iota_sdk/types/unlock_condition.py]: Just curious, where are you taking docs from?

**Review by Thoralf-M — DISMISSED** [review]

**Review by DaughterOfMars — DISMISSED** [review]

File: bindings/python/iota_sdk/types/unlock_condition.py (hunk 1 above)
Comment (thibault-martinez) [reply bindings/python/iota_sdk/types/unlock_condition.py]: Would be nice to always have the same, if possible coming from the TIP (not API)

Comment (kwek20) [reply bindings/python/iota_sdk/types/unlock_condition.py]: I took from the rust side(as in ExpirationUnlockCondition), guess i shouldve checked for changes..again :D

Comment (kwek20) [reply bindings/python/iota_sdk/types/unlock_condition.py]: fixed now 👍

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Format

File: bindings/python/iota_sdk/types/unlock_condition.py (hunk 1 above)
Comment (thibault-martinez) [reply bindings/python/iota_sdk/types/unlock_condition.py]: It's still not the same as in the TIP? Also this doc is wrong to start with, this is not expiration.
Take doc from https://github.com/iotaledger/tips-draft/blob/tip38/tips/TIP-0038/tip-0038.md#timelock-unlock-condition

**Review by Alex6323 — DISMISSED** [review]

**Review by DaughterOfMars — DISMISSED** [review]

**Review by DaughterOfMars — DISMISSED** [review]

**Review by Thoralf-M — DISMISSED** [review]

File: bindings/python/iota_sdk/types/unlock_condition.py (hunk 1 above)
Comment (thibault-martinez) [reply bindings/python/iota_sdk/types/unlock_condition.py]: Still wrong

File: bindings/python/iota_sdk/types/unlock_condition.py [hunk 2]
```diff
@@ -83,7 +83,7 @@ class TimelockUnlockCondition:
 @json
 @dataclass
 class ExpirationUnlockCondition:
-    """An expiration unlock condition.
+    """Defines a slot index until which only Address, defined in Address Unlock Condition, is allowed to unlock the output. After the slot index is reached/passed, only Return Address can unlock it.
```
Comment (DaughterOfMars) [inline bindings/python/iota_sdk/types/unlock_condition.py]:
```suggestion
    """Defines a slot index until which only the Address defined in the Address Unlock Condition is allowed to unlock the output. After the slot index is reached/passed, only the Return Address can unlock it.
```

Comment (kwek20) [reply bindings/python/iota_sdk/types/unlock_condition.py]: suggested it in tip as well

File: bindings/nodejs/lib/types/block/output/unlock-condition.ts [hunk 3]
```diff
@@ -156,8 +156,8 @@ class TimelockUnlockCondition extends UnlockCondition {
 }
 
 /**
- * Defines an expiration slot index. Before the slot index is reached, only the Address defined in the Address
- * Unlock Condition is allowed to unlock the output. Afterward, only the Return Address can unlock it.
+ * Defines a slot index until which only Address, defined in Address Unlock Condition, is allowed to unlock the output.
+ * After the slot index is reached/passed, only Return Address can unlock it.
```
Comment (Alex6323) [inline bindings/nodejs/lib/types/block/output/unlock-condition.ts]:
```suggestion
 * Defines an unlock condition that binds an expiry time/slot index to the address given in the Address Unlock 
 * Condition of the same output. Once that address has expired, only the address defined in `returnAddress` can 
 * unlock the output. 
```

Comment (kwek20) [reply bindings/nodejs/lib/types/block/output/unlock-condition.ts:160]: i suggested the current line in the tip document. unless you feel like your suggestion is better?

**Comment by thibault-martinez** [comment]
The issue was fixed by another PR, is this still relevant?

**Review by Thoralf-M — DISMISSED** [review]

**Comment by Thoralf-M** [comment]
> The issue was fixed by another PR, is this still relevant?

Yes some things weren't changed

File: bindings/nodejs/lib/types/block/output/unlock-condition.ts [hunk 4]
```diff
@@ -168,8 +168,7 @@ class ExpirationUnlockCondition extends UnlockCondition {
     })
     readonly returnAddress: Address;
     /**
-     * Before this slot index, the condition is allowed to unlock the output,
-     * after that only the address defined in return address.
+     * Before this slot index, Address Unlock Condition is allowed to unlock the output, after that only the address defined in Return Address.
```
Comment (DaughterOfMars) [inline bindings/nodejs/lib/types/block/output/unlock-condition.ts]:
```suggestion
     * Before this slot index, the Address Unlock Condition is allowed to unlock the output, after that only the address defined in Return Address can.
```

File: bindings/python/iota_sdk/types/unlock_condition.py [hunk 5]
```diff
@@ -71,11 +71,11 @@ class StorageDepositReturnUnlockCondition:
 @json
 @dataclass
 class TimelockUnlockCondition:
-    """A timelock unlock condition.
+    """Defines a slot index until which the output can not be unlocked.
     Args:
-        unix_time: The Unix timestamp marking the end of the timelock.
+        slot_index: Slot index starting from which the output can be consumed.
```
Comment (DaughterOfMars) [inline bindings/python/iota_sdk/types/unlock_condition.py]:
```suggestion
        slot_index: Slot index that defines when the output can be consumed.
```

**Review by DaughterOfMars — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**